[Feature] New mechanism for preIPO assets

csanti opened the pull request:
### Overview

We implement a new mechanism that prevents having to migrate from preIPO asset to postIPO asset and allow preIPO traders to hold the assets without having to burn and re-mint new assets.

Instead of relying on asset migration to mark the event of an IPO, we add a new mint contract operation `TriggerIPO` to make the transition from preIPO to postIPO. 

Asset lifecycle:
 1. Asset is whitelisted through factory contract, filling `mint_period` parameter and `min_collateral_ratio_after_ipo` parameters.
```
pub struct Params {
    /// Auction discount rate applied to asset mint
    pub auction_discount: Decimal,
    /// Minium collateral ratio applied to asset mint
    pub min_collateral_ratio: Decimal,
    /// Distribution weight (default is 30, which is 1/10 of MIR distribution weight)
    pub weight: Option<u32>,
    /// For pre-IPO assets, time period after asset creation in which minting is enabled
    pub mint_period: Option<u64>,
    /// For pre-IPO assets, collateral ratio for the asset after ipo
    pub min_collateral_ratio_after_ipo: Option<Decimal>,
}
```
2. Asset is registered to mint_contract, with `ipo_params`:
```
RegisterAsset {
    asset_token: HumanAddr,
    auction_discount: Decimal,
    min_collateral_ratio: Decimal,
    ipo_params: Option<IPOParams>,
},
pub struct IPOParams {
    pub mint_end: u64,
    pub min_collateral_ratio_after_ipo: Decimal,
}
```
3. When IPO event happens, feeder executes `TriggerIPO` operation and starts feeding real price. This operation will assign a new `min_cr` to the asset, making the asset a traditional asset without `mint_period`.
```
TriggerIPO {
    asset_token: HumanAddr, 
},
```

### Behaviour summary

- During mint_period:
   - All operations enabled
   - Feeder feeds fixed price
- After mint_period (Trading phase):
   - Mint and burn disabled
   - Feeder keeps feeding fixed price
- After IPO event:
   - `min_cr` is updated
   - Feeder executes `TriggerIPO`
   - Feeder starts feeding real price 

*Migration of preIPO asset is not contemplated

YunSuk-Yeo reviewed (changes requested):
Seems good to use time units instead height for pre-ipo mint & burn period

other things are LGTM

csanti commented:
Added the following changes:
- mint_period using time instead of height
- during mint_period fixed price is stored in mint contract instead of relying on fixed oracle price feed
- added requested logs for preIPO whitelisting

csanti commented on contracts/mirror_mint/src/positions.rs:
```diff
@@ -580,10 +579,8 @@ pub fn burn<S: Storage, A: Api, Q: Querier>(
-    if asset_config.end_price.is_some() {
-        let oracle: HumanAddr = deps.api.human_address(&config.oracle)?;
-        let asset_price: Decimal =
-            load_asset_price(&deps, &oracle, &position.asset.info, Some(env.block.time))?;
```
We don't need it anymore because now we query `asset_price` and `collateral_price` separately.
